website/docs/cloud-docs/workspaces/dynamic-provider-credentials/hcp-configuration.mdx

```diff
@@ -0,0 +1,110 @@
+---
+page_title: Dynamic Credentials with the HCP Provider - Workspaces - HCP Terraform
+description: >-
+  Use OpenID Connect to get short-term credentials for the HCP Terraform provider in
+  your HCP Terraform runs.
+---
+
+# Dynamic Credentials with the HCP Provider
+
+~> **Important:** If you are self-hosting [Terraform Cloud agents](/terraform/cloud-docs/agents), ensure your agents use [v1.15.1](/terraform/cloud-docs/agents/changelog#1-15-1-05-01-2024) or above. To use the latest dynamic credentials features, [upgrade your agents to the latest version](/terraform/cloud-docs/agents/changelog).
+
+You can use HCP Terraform’s native OpenID Connect integration with HCP to get [dynamic credentials](/terraform/cloud-docs/workspaces/dynamic-provider-credentials) for the HCP provider in your HCP Terraform runs. Configuring the integration requires the following steps:
+
+1. **[Configure HCP](#configure-hcp):** Set up a trust configuration between HCP and HCP Terraform. Then, you must create an [service principal in HPC](https://developer.hashicorp.com/hcp/docs/hcp/admin/iam/service-principals) for your HCP Terraform workspaces.
+2. **[Configure HCP Terraform](#configure-hcp-terraform):** Add environment variables to the HCP Terraform workspaces where you want to use Dynamic Credentials.
+
+Once you complete the setup, HCP Terraform automatically authenticates to HCP during each run.
+
+## Configure HCP
+You must enable and configure a workload identity pool and provider on HCP. These instructions use the HCP CLI, but you can also use Terraform to configure HCP. Refer to our [example Terraform configuration](https://github.com/hashicorp/terraform-dynamic-credentials-setup-examples/tree/main/hcp).
+
+#### Create a Service Principal
+You will need to create a service principal for HCP Terraform to assume during runs. Remember the ID of the service principal as you will need it for the next steps.
+
+```shell
+hcp iam service-principals create hcp-terraform --project=HCP_PROJECT_ID
```

Comment:
Do we need to call out that they need to create this service principal within a certain HCP project? Maybe link to where they can figure out their HCP project's ID?

Reply:
Yeah, good callout 🤔 I added a bit more detail around the project ID. I'm not sure we need to tell users how to get the project ID in this document though, I went and looked a few other examples of documentation from other products (i.e. [this one](https://developer.hashicorp.com/packer/docs/hcp#hcp_project_id) from Packer) that mentions the HCP project ID and it looks they don't go into too much detail either, which makes me believe HCP Project IDs are a basic enough concept that folks can find them easily enough.

Reply:
Love it! Thank you for adding the perfect amount of detail here. How would you feel about moving the sentences you wrote about the HCP project ID under the code block? Moving them around in this order:

```
Create a service principal for HCP Terraform to assume during runs by running the following HCP command. Note the ID of the service principal you create because you will need it in the following steps. 

hcp iam service-principals create hcp-terraform --project=HCP_PROJECT_ID

For all remaining steps, replace `HCP_PROJECT_ID` with the ID of the project that contains all the resources and workspaces that you want to manage with this service principal....etc.
```

That way folks get the command _and_ have the direct context for where to use the `HCP_PROJECT_ID`.